edb/edgeql/parser/__init__.py

```diff
@@ -188,3 +181,72 @@ def preload(
                 pool.map(_load_parser, parsers_to_rebuild)
 
             preload(parsers=parsers, allow_rebuild=False)
+
+
+def process_spec(parser: parsing.ParserSpec) -> Tuple[str, List[Any]]:
+    # Converts a ParserSpec into JSON. Called from edgeql-parser Rust crate.
+
+    spec = parser.get_parser_spec()
+    assert spec.pureLR
+
+    rmap = {v._token: c for (_, c), v in parsing.TokenMeta.token_map.items()}
+
+    # productions
+    productions: List[Any] = []
+    production_ids: Dict[Any, int] = {}
+    inlines: List[Tuple[int, int]] = []
+
+    def get_production_id(prod: Any) -> int:
+        if prod in production_ids:
+            return production_ids[prod]
+
+        id = len(productions)
+        productions.append(prod)
+        production_ids[prod] = id
+
+        inline = getattr(prod.method, 'inline_index', None)
+        if inline is not None:
+            assert isinstance(inline, int)
+            inlines.append((id, inline))
+
+        return id
+
+    # XXX: TOKENS
```

Comment:
that's your comment I don't know what you meant

Reply:
Yeah, neither do I really. Maybe it meant that we should be doing something so that the token lookups in the parser weren't string based?
I don't know.